File: mobile/shared/src/commonMain/kotlin/com/bissbilanz/model/Recipe.kt
package com.bissbilanz.model

import kotlinx.serialization.Serializable

@Serializable
data class Recipe(
    val id: String,
    val userId: String,
    val name: String,
    val totalServings: Double,
    val isFavorite: Boolean = false,
    val imageUrl: String? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null,
    val ingredients: List<RecipeIngredient>? = null,
)

@Serializable
data class RecipeIngredient(
    val id: String? = null,
    val recipeId: String? = null,
    val foodId: String,
    val quantity: Double,
    val servingUnit: ServingUnit,
    val sortOrder: Int = 0,
    val food: Food? = null,
)

@Serializable
data class RecipeCreate(
    val name: String,
    val totalServings: Double,
    val ingredients: List<RecipeIngredientInput>,
    val isFavorite: Boolean? = null,
    val imageUrl: String? = null,
)

@Serializable
data class RecipeIngredientInput(
    val foodId: String,
    val quantity: Double,
    val servingUnit: ServingUnit,
)

@Serializable
data class RecipeUpdate(
    val name: String? = null,
    val totalServings: Double? = null,
    val ingredients: List<RecipeIngredientInput>? = null,
    val isFavorite: Boolean? = null,
    val imageUrl: String? = null,
)

File: mobile/iosApp/Bissbilanz/Models/Recipe.swift
import Foundation

struct Recipe: Codable, Identifiable, Hashable {
    let id: String
    let userId: String
    let name: String
    let totalServings: Double
    let isFavorite: Bool
    let imageUrl: String?
    let calories: Double?
    let protein: Double?
    let carbs: Double?
    let fat: Double?
    let fiber: Double?
    let createdAt: String?
    let updatedAt: String?
    let ingredients: [RecipeIngredient]?

    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }

    static func == (lhs: Recipe, rhs: Recipe) -> Bool {
        lhs.id == rhs.id
    }
}

struct RecipeIngredient: Codable, Identifiable {
    let id: String?
    let recipeId: String?
    let foodId: String
    let quantity: Double
    let servingUnit: ServingUnit
    let sortOrder: Int
    let food: Food?
}

struct RecipeCreate: Codable {
    let name: String
    let totalServings: Double
    let ingredients: [RecipeIngredientInput]
    var isFavorite: Bool?
    var imageUrl: String?
}

struct RecipeIngredientInput: Codable {
    let foodId: String
    let quantity: Double
    let servingUnit: ServingUnit
}

struct RecipeUpdate: Codable {
    var name: String?
    var totalServings: Double?
    var ingredients: [RecipeIngredientInput]?
    var isFavorite: Bool?
    var imageUrl: String?
}

struct RecipesResponse: Codable {
    let recipes: [Recipe]
}

struct RecipeResponse: Codable {
    let recipe: Recipe
}
